e2e/cypress/support/saml_commands.js

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import * as TIMEOUTS from '../fixtures/timeouts';
+
+Cypress.Commands.add('checkLoginPage', (settings = {}) => {
+    // * Check the title
+    cy.title().should('include', settings.siteName);
+
+    // * Check elements in the body
+    cy.get('#loginId').should('be.visible').and(($loginTextbox) => {
+        const placeholder = $loginTextbox[0].placeholder;
+        expect(placeholder).to.match(/Email/);
+        expect(placeholder).to.match(/Username/);
+    });
+    cy.get('#loginPassword').should('be.visible').and('have.attr', 'placeholder', 'Password');
+    cy.findByText('Sign in').should('be.visible');
+});
+
+Cypress.Commands.add('checkSignUpPageSection', () => {
+    cy.findByText('Your guest account has no channels assigned. Please contact an administrator.').should('be.visible');
+});
+
+Cypress.Commands.add('checkChatSideBar', (settings = {}) => {
+    if (settings.teamName != null && settings.teamName.length > 0) {
+        cy.get('#headerTeamName').should('be.visible').and('contain', settings.teamName);
+    }
+
+    if (settings.user.username.length > 0) {
+        cy.get('#headerUsername').should('be.visible').and('contain', settings.user.username);
+    }
+
+    if (settings.user.userType === 'Admin' || settings.user.isAdmin) {
+        //check that he is an admin
+        cy.get('#sidebarHeaderDropdownButton').click().then(() => {
+            cy.findByText('System Console').should('be.visible');
+            cy.get('#sidebarHeaderDropdownButton').click();
+        });
+    } else {
+        cy.get('#sidebarHeaderDropdownButton').click().then(() => {
+            cy.findByText('System Console').should('not.be.visible');
+            cy.get('#sidebarHeaderDropdownButton').click();
+        });
+    }
+    cy.get('#channel_view').should('be.visible');
+});
+
+Cypress.Commands.add('checkInvitePeoplePage', (settings = {}) => {
+    if (settings.teamName != null && settings.teamName.length > 0) {
+        cy.findByText('Invite people to ' + settings.teamName).should('be.visible');
+    }
+    cy.findByText('Copy Link').should('be.visible');
+});
+
+Cypress.Commands.add('checkCreateTeamPage', (settings = {}) => {
+    if (settings.user.userType === 'Guest' || settings.user.isGuest) {
+        cy.findByText('Create a team').should('not.be.visible');
+    } else {
+        cy.findByText('Create a team').should('be.visible');
```

Comment:
This step fails if there are a lot of teams and the option to Create a Team is displayed at the bottom. In this case it's not visible. Hence to overcome this, please update it to something like this:
`cy.findByText('Create a team').scrollIntoView().should('be.visible');`

Reply:
Seen that too.